addons/website/static/src/js/content/snippets.animation.js

```diff
@@ -1050,9 +1050,14 @@ registry.ScrollButton = registry.anchorSlide.extend({
      */
     _onAnimateClick: function (ev) {
         ev.preventDefault();
-        const $nextElement = this.$el.closest('section').next();
-        if ($nextElement.length) {
-            this._scrollTo($nextElement);
+        // Scroll to the first visible section after the current section.
+        const currentSectionEl = this.$target[0].closest('section');
+        const sectionEls = [...currentSectionEl.parentNode.querySelectorAll('section')];
+        const currentSectionIndex = sectionEls.indexOf(currentSectionEl);
+        const nextSectionEls = sectionEls.slice(currentSectionIndex + 1);
```

Comment:
Those are two "expensive" operation you do not have to do. In this case, the most readable way to implement this is the most efficient way... just loop through next elements (using `nextElementSibling`) until you find a visible one.